### DISCUSSION TITLE
Generating multiple serializers for the same types

### DISCUSSION THREAD

**Discussion by Astn** [jamescourtney/FlatSharp#352] [Q&A] [answered]
@jamescourtney 
There are some very user friendly features with some of the serializer options like `lazy`, but in some situations another mode works better. 

I'm thinking of having multiple `.fbs` files with the serializer annotations specified differently in each of the files, but all of the data types identical.

Then depending on which type of serialization I want to use, I can easily have access to the other serializers.

Does this seem like a good approach?

**Comment by jamescourtney** [comment, marked as answer]
FlatSharp v7 does this by default: https://github.com/jamescourtney/FlatSharp/releases/tag/7.0.0. The `fs_serializer` attribute now only specifies the default serializer for `YourType.Serializer`. There are a few ways to gain access to the other serializers:

- `YourType.Serializer.Parse(buffer, FlatBufferDeserializationOption.Progressive)`
- `var progressiveSerializer = YourType.Serializer.WithSettings(opts => opts.UseProgressiveDeserialization());`
- If on .NET 7:
```c#
public static T ParseProgressive<T>(byte[] buffer) where T : IFlatBufferSerializable<T>
{
    return T.ProgressiveSerializer.Parse(buffer);
}
```
